shared/domains/chelonia/chelonia.js

```diff
@@ -1198,6 +1200,58 @@ export default (sbp('sbp/selectors/register', {
     })
     return msg
   },
+  'chelonia/out/ownResources': async function (contractID: string) {
+    if (!contractID) {
+      throw new TypeError('A contract ID must be provided')
+    }
+
+    const response = await fetch(`${this.config.connectionURL}/ownResources`, {
+      method: 'GET',
+      signal: this.abortController.signal,
+      headers: new Headers([
+        [
+          'authorization',
+          buildShelterAuthorizationHeader.call(this, contractID)
+        ]
+      ])
+    })
+    if (!response.ok) {
+      throw new Error(`Unable to fetch own resources for ${contractID}`)
+    }
+
+    return response.json()
+  },
+  'chelonia/out/deleteContract': async function (contractID: string | string[], credentials: { [contractID: string]: { token: ?string, billableContractID: ?string } } = {}) {
+    if (!contractID) {
+      throw new TypeError('A contract ID must be provided')
+    }
+    if (!Array.isArray(contractID)) contractID = [contractID]
+    return await Promise.allSettled(contractID.map(async (cid) => {
+      const hasCredential = has(credentials, cid)
+      const hasToken = has(credentials[cid], 'token') && credentials[cid].token
+      const hasBillableContractID = has(credentials[cid], 'billableContractID') && credentials[cid].billableContractID
+      if (!hasCredential || (!hasToken && hasToken === hasBillableContractID)) {
+        throw new TypeError(`Either a token or a billable contract ID must be provided for ${cid}`)
+      }
```

Comment:
Seems more straightforward to write:

```js
if (!hasCredential || (!hasToken && !hasBillableContractID)) { … }
```

Reply:
That's not the relly same, as the intent is that this should be either-or (i.e., XOR). However, I think that `(!hasCredential || hasToken === hasBillableContractID)` would work.